File: android/beaconsdk/src/main/java/com/xenber/beaconsdk/BeaconConfig.kt
package com.xenber.beaconsdk

data class BeaconConfig (

  val gatewayUrl: String,
  val dataUrl: String,
  val userId: String,
  //val authToken: String, 
  val rssiThreshold: Int = -85,
  val timeThreshold: Int = 2,
  val scanPeriod: Long = 1100L,
  val betweenScanPeriod: Long = 5000L,
  val notificationCooldown: Long = 3000L,
  val maxNotifications: Int = 3

) {

  init {
    require(gatewayUrl.startsWith("http")) {"Invalid gateway URL"}
    require(dataUrl.startsWith("http")) {"Invalid data URL"}
    require(userId.isNotBlank()) {"User ID Missing"}
    //require(authToken.isNotBlank()) {"Auth Token Missing"}
  }
}

File: ios/Runner/BeaconSdk/Core/BeaconConfig.swift
import Foundation

public struct BeaconConfig {
    //public var gatewayUrl: String
    public var dataUrl: String
    public var userId: String
    public var rssiThreshold: Int
    public var timeThreshold: Int 
    public var scanPeriod: Int 
    public var betweenScanPeriod: Int
    public var notificationCooldown: Int 

    public init(
        dataUrl: String,
        userId: String,
        rssiThreshold: Int = -85,
        timeThreshold: Int = 2,
        scanPeriod: Int = 1100,
        betweenScanPeriod: Int = 5000,
        notificationCooldown: Int = 60000
    ) {
        self.dataUrl = dataUrl
        self.userId = userId
        self.rssiThreshold = rssiThreshold
        self.timeThreshold = timeThreshold
        self.scanPeriod = scanPeriod
        self.betweenScanPeriod = betweenScanPeriod
        self.notificationCooldown = notificationCooldown
    }
}